python/triton/compiler/compiler.py

```diff
@@ -244,8 +244,8 @@ def compile(src, target=None, options=None):
     # core changes to make it easier to track kernels by hash.
     enable_override = os.environ.get("TRITON_KERNEL_OVERRIDE", "0") == "1"
     enable_ir_dump = os.environ.get("TRITON_KERNEL_DUMP", "0") == "1"
-    fn_override_manager = get_override_manager(src.hash()) if enable_override else None
-    fn_dump_manager = get_dump_manager(src.hash()) if enable_ir_dump else None
+    fn_override_manager = get_override_manager(hash) if enable_override else None
+    fn_dump_manager = get_dump_manager(hash) if enable_ir_dump else None
```

Comment:
the hash are different on purpose as the caching hash changes when triton changes while we don't want that for debug purpose since it makes it much harder to identify kernels.
What is the motivation for this change?

Reply:
I see. I was using `TRITON_KERNEL_OVERRIDE=1` based on the cache. I got the IRs from the cache, modified it and stored in the override directory with same hash. But I noticed that the compiler didn't pick it up, as it looked for a different hash. 

How do you use `TRITON_KERNEL_OVERRIDE` usually?

Reply:
ah yeah you want to use it with what is dumped from `TRITON_KERNEL_DUMP`. This will dump files in ~/.triton/dump/ then you can copy those files with the same hash into override folder.
Otherwise any change to triton invalidates the hash which makes using override quite painful

Reply:
I see. BTW, I'd like the override folder to be specified by an env var so that I don't have to clean the default dump/override folder when it gets too big. WDYT?

Reply:
yeah I guess it is an option (others have also asked that). Would it be better if the default was in `/tmp/triton/override`?

Reply:
We work in a shared dev server environment so `/tmp/triton/override` may cause a conflict potentially. I think something like `TRITON_CACHE_DIR` for dump or override is helpful, especially when working on multiple kernels at same time.

Reply:
Something like https://github.com/triton-lang/triton/pull/4282. WDYT?

Reply:
sounds good